robot/basestation/app.py

```diff
@@ -1,168 +1,206 @@
 #!/usr/bin/env python3
-# This is the controller page of the Flask GUI
-# Flask is light-weight and modular so this is actually all we need to set up a simple HTML page
+"""This is the controller page of the Flask GUI
+Flask is light-weight and modular so this is actually all we need to set up a simple HTML page"""
 
+import re
+import os
+import subprocess
 import flask
+from flask import jsonify
 
 app = flask.Flask(__name__)
 
-# set the rover's IP
-# rover static ip (tenatative): 192.168.1.20
-ROVER_IP = "127.31.43.134"
+# get rover's IP from global bash environment
+rover_ip = os.environ["ROS_MASTER_URI"]
+rover_ip = re.findall(r"(.*):\d+", rover_ip)[0]
+#print("rover_ip: " + rover_ip)
 
-# Once we launch this, this will route us to the "../" page or index page and
-# automatically render the Rover GUI
+rover_ip_raw = rover_ip.split("//")[1]
+
+def run_bash(cmd):
+    """Runs bash command supplied as string,
+    executes it and returns tuple of output and error
+    """
+    process = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+
+    return output, error
+
+# Once we launch this, this will route us to the "/" page or index page and
+# automatically render the Robot GUI
 @app.route("/")
 def index():
-    return flask.render_template("AsimovOperation.html", roverIP=ROVER_IP)
+    """Current Landing page, the arm panel"""
+
+    global rover_ip, rover_ip_raw
+
+    # get rover's IP from global bash environment
+    rover_ip = os.environ["ROS_MASTER_URI"]
+    rover_ip = re.findall(r"(.*):\d+", rover_ip)[0]
+    #print("rover_ip: " + rover_ip)
+
+    rover_ip_raw = rover_ip.split("//")[1]
```

Comment:
Why are you redefining `rover_ip_raw` on this function if you define it in its outer scope anyways?

Reply:
re-setting it in case it was changed in the meantime. Like if the dev was testing with the odroid, then changed to local.

Reply:
I'm still not sure I see why it would change during the execution of the code, it smells dirty. But regardless, at the very least why not abstract the fetching and parsing of the ip value in a function and instead call that function which returns `rover_ip, rover_ip_raw` and use it accordingly. I'll make a commit soon to showcase this along with the other comment about setting up the direnv stuff.